### PR TITLE
RISC-V kde-common recipe missing FTBFS packages survey 20240326

### PR DESCRIPTION
**Pull request by Icenowy**
Topic Description
-----------------

- thunderbird: support build for riscv64
    - Drop --disable-elf-hack because it's not available on riscv64 either.
    - Added a patch to fix linking issue.
    Signed-off-by: Icenowy Zheng <uwu@icenowy.me>
- libreoffice: support build for riscv64
    - Inject a patch to firebird like what is done on LoongArch.
    Signed-off-by: Icenowy Zheng <uwu@icenowy.me>
- thunderbird: fix patch applying issue
    Currently one patch could not be cleanly applied.
    Manually fix it.
    Signed-off-by: Icenowy Zheng <uwu@icenowy.me>
- libreoffice: fix patch applying issue
    Currently one patch could not be cleanly applied.
    Manually fix it.
    Signed-off-by: Icenowy Zheng <uwu@icenowy.me>
- foo2zjs: fix patch applying issue
    Currently one patch could not be cleanly applied.
    Manually fix it.
    Signed-off-by: Icenowy Zheng <uwu@icenowy.me>

Package(s) Affected
-------------------

- foo2zjs: 20210115-1
- libreoffice: 7.5.4.2-3
- thunderbird: 115.5.2-1

Security Update?
----------------

No

Build Order
-----------

```
#buildit foo2zjs libreoffice thunderbird
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [x] AArch64 `arm64`
- [ ] LoongArch 64-bit `loongarch64`

**Secondary Architectures**

- [ ] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

**Experimental Architectures**

- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`

### CODE REVIEW

File: app-productivity/libreoffice/autobuild/build [hunk 1]
```diff
@@ -96,14 +96,13 @@ abinfo "Autogen-ing ..."
              --disable-dependency-tracking \
              --enable-pch=full
 
-if ! ab_match_arch powerpc && \
-   ! ab_match_arch ppc64 && \
-   ! ab_match_arch loongson3; then
+if ab_match_arch loongarch64 || \
+   ab_match_arch riscv64; then
```
Comment (jiegec) [inline app-productivity/libreoffice/autobuild/build:100]: Missing amd64/arm64?

Comment (jiegec) [reply app-productivity/libreoffice/autobuild/build:100]: The patch is for loongarch64 only, but is firebird required on amd64/arm64?

Comment (Icenowy) [reply app-productivity/libreoffice/autobuild/build:100]: This is for preloading the source code to patch.

Firebird is required everywhere, but this procedure is for patching the source.

On other architectures its source code will be downloaded automatically.

**Review by jiegec — APPROVED** [review]
LGTM

**Pull request merged**